src/dmodule.d

```diff
@@ -1068,15 +1070,17 @@ public:
         return needmoduleinfo || global.params.cov;
     }
 
-    override Dsymbol search(Loc loc, Identifier ident, int flags = IgnoreNone)
+    override Dsymbol search(Loc loc, Identifier ident, int flags = IgnoreImports)
     {
         /* Since modules can be circularly referenced,
          * need to stop infinite recursive searches.
          * This is done with the cache.
          */
-        //printf("%s Module::search('%s', flags = %d) insearch = %d\n", toChars(), ident->toChars(), flags, insearch);
+        //printf("%s Module.search('%s', flags = x%x) insearch = %d\n", toChars(), ident.toChars(), flags, insearch);
         if (insearch)
             return null;
+        if (!(flags & IgnoreRoot))
+            flags &= ~(IgnoreRoot | IgnoreImports);          // searching an import is always transitive
```

Comment:
I know what transitive means, it's just very hard to relate the comment to `if (!(flags & IgnoreRoot)) flags &= ~(IgnoreRoot | IgnoreImports)`.
As this seems to be the core mechanism of the PR, a few more words would be adequate.